update_workbook.py

```diff
@@ -0,0 +1,238 @@
+    SCIPP = auto()
+
+lecture_to_directories = MappingProxyType({
+    Lecture.MCSTAS: Path('mcstas'),
+    Lecture.MODEL: Path('model'),
+    Lecture.PROPOSALS: Path('proposals'),
+    Lecture.PYTHONBASICS: Path('python'),
+    Lecture.SCICAT: Path('scicat'),
+    Lecture.SCIPP: Path('scipp'),
+})
+
+# Helper Utilities
+# Helper - Git
+def run_git_command(*args: str) -> str:
+    import subprocess
+    if 'rm' in args and '--cached' not in args:
+        raise RuntimeError("Using `git rm` without --cached option is not allowed.")
+
+    return subprocess.run(['git', *args], stdout=subprocess.PIPE, text=True).stdout
+
+
+def git_changed_files(git_base: str = './') -> dict[Path, GitStatus]:
+    git_status = run_git_command('-C', git_base, 'status', '-s')
+    return {
+        Path(file_status[3:]): git_prefix_to_status[status]
+        for file_status in git_status.split('\n')
+        if (status:=file_status[:3]) in git_prefix_to_status
+    }
+
+
+def git_root() -> Path:
+    git_root = run_git_command('rev-parse', '--show-toplevel')
+    return Path(git_root.rstrip())
+
+
+def report_updates() -> None:
+    changed_list = git_changed_files(git_base=str(WORKBOOK_ROOT_DIR))
+
+    if changed_list:
+        print("Updates done.")
+        print("There are updates in the workbooks to be committed.")
+        
+        print("\nHint:\n",
+            "cd workbooks\n",
+            *(f"git add {str(filepath)}\n" for filepath in changed_list),
+            "git commit -m '{COMMITMESSAGE}'\n",
+            "git push origin {BRANCHNAME}")
+    else:
+        print("Nothing to update.")
+
+
+# Helper - System Commands
+def check_current_dir():
+    import os
+    if (cur_file_dir:=Path(__file__).parent) != git_root() \
+        or cur_file_dir != Path(os.getcwd()):
+        raise RuntimeError("`update_workbook` should be run in "
+                           "the `dmsc-school` root directory, where this script is."
+                           f"hint: cd {cur_file_dir.resolve()}")
+
+
+def listmaterials(root_dir: Path) -> list:
+    import os
+    materials = []
+    for lecture_dir in lecture_to_directories.values():
+        try:
+            file_paths = os.listdir(root_dir/lecture_dir)
+            materials.extend([lecture_dir/Path(file_path) for file_path in file_paths])
+        except FileNotFoundError:
+            ...
+    
+    return materials
+
+
+def get_all_materials() -> dict:
+    textbook_list = listmaterials(root_dir=Path('./'))
+    workbook_list = listmaterials(root_dir=WORKBOOK_ROOT_DIR)
+
+    materials = dict()
+    for workbook in workbook_list:
+        if workbook not in textbook_list:
+            materials[workbook] = GitStatus.DELETED
+
+    for textbook in textbook_list:
+        materials[textbook] = GitStatus.TOUCHED
+
+    return materials
+
+def export_requirements() -> None:
+    """Copy requirements.txt to the workbook directory."""
+    import shutil
+    shutil.copy("requirements.txt", WORKBOOK_ROOT_DIR)
```

Comment:
@jl-wynen  I also added one more thing to copy `requirements.txt` into workbook. 
Will you give this comment thumbs up if it looks fine? : D

Reply:
Do we still use `requirements.txt`? I thought we switched entirely to conda?

Reply:
So are we going to remove `requirements.txt` for good...?

Reply:
I thought so. Or is it still required for jupyter book?

Reply:
It is required for the Jupyter Book to build on the current GitHub action.

Reply:
Can you change it to use conda? We shouldn't duplicate the dependencies

Reply:
Regardless of that, I removed it. Thanks!

Reply:
Removed the `requirements.txt`?

Reply:
I guess it is just a matter of changing the workflow? https://github.com/ess-dmsc-dram/dmsc-school/blob/64023474282a8215921439d8c56ec1ad3b4a40b6/.github/workflows/deploy-book.yml#L26